export/export.py

```diff
@@ -0,0 +1,866 @@
+        # But we use generic parameter names here for simplicity
+        self._stages[stage_name].run(primary_input, config_params)
+
+        # Return the outputs
+        return self._stages[stage_name].get_outputs()
+
+    def _get_stage_inputs(self, stage_name: str) -> tuple[Any, Dict[str, Any]]:
+        """
+        Get all inputs for a specific stage.
+
+        Args:
+            stage_name: Name of the stage to get inputs for
+
+        Returns:
+            A tuple containing:
+            - The primary input for the stage
+            - Dictionary of configuration parameters for the stage
+        """
+        if stage_name == "source_transform":
+            # For SourceTransformStage.run(models, **kwargs)
+            return self._model, {}
+
+        elif stage_name == "quantize":
+            # For QuantizeStage.run(model_data, calibration_config)
+            model_data = {
+                "model": self._model,
+            }
+            calibration_config = {
+                "example_inputs": self._example_inputs,
+            }
+            return model_data, calibration_config
+
+        elif stage_name == "export":
+            # For ExportStage.run(models, export_config)
+            models = {
+                "model": self._model,
+            }
+            export_config = {
+                "example_inputs": self._example_inputs,
+                "dynamic_shapes": self._dynamic_shapes,
+            }
+            return models, export_config
+
+        elif stage_name == "edge_transform_and_lower":
+            # For EdgeTransformAndLowerStage.run(exported_programs, transform_config)
+            exported_programs = self._exported_program
+            transform_config = {
+                "constant_methods": self._constant_methods,
+            }
+            return exported_programs, transform_config
+
+        elif stage_name == "executorch":
+            # For ExecutorchStage.run(edge_program, backend_options)
+            edge_program = self._edge_program_manager
+            backend_options = {}
+            return edge_program, backend_options
+
+        else:
+            return None, {}
+
+    def _store_stage_result(self, stage_name: str, result: Any) -> None:
+        """
+        Store the result of a stage.
+
+        Args:
+            stage_name: Name of the stage
+            result: Result of the stage
+        """
+        if stage_name == "source_transform":
+            self._model = result
+        elif stage_name == "quantize":
+            self._model = result
+        elif stage_name == "export":
+            self._exported_program = result
+        elif stage_name == "edge_transform_and_lower":
+            self._edge_program_manager = result
+            self._delegation_info = self._stages[stage_name].delegation_info
+        elif stage_name == "executorch":
+            self._executorch_program_manager = result
+
+    def _run_pipeline(self, start_stage: str) -> None:
```

Comment:
Why do we need this start_stage stuff, can we just not take any parameters and just run the all stages in self._stages?

Reply:
Yeah I agree. We should not have this start_stage thing. We should just construct pipeline that can just be run with each stage in sequence. Construction of the pipeline should take care of apprpriate deps

Reply:
Addressed in latest version.